File: RR.Common/Models/ReceiptItem.cs
namespace RR.Common.Models;

public class ReceiptItem
{
    public string? Name { get; set; }
    public Product? Product { get; set; }
    public float Quantity { get; set; }
    public Measurement Measurement { get; set; }
    public float PricePerQuantity { get; set; }
    public float Price => Quantity * PricePerQuantity;
}

public class ReceiptItemValidated : ReceiptItem
{
    public Dictionary<string, string>? ValidationMessages { get; set; }
}

File: RR.Frontend/src/DTOs/ReceiptItem.ts
import type { Measurement } from './Measurement'

export class ReceiptItem {
  originalRecognizedName?: string;
  name: string;
  quantity: number;
  measurement: Measurement;
  pricePerQuantity: number;
  price: number;

  constructor() {
    this.name = '';
    this.quantity = 0;
    this.measurement = 'pieces';
    this.pricePerQuantity = 0;
    this.price = 0;
  }

  static from(obj: any): ReceiptItem {
    return Object.assign(new ReceiptItem(), obj);
  }
  
  isEmpty(): boolean {
    return (!this.name || this.name.trim() === '') &&
      (!this.originalRecognizedName || this.originalRecognizedName.trim() === '') &&
      (!this.quantity || this.quantity === 0) &&
      (!this.pricePerQuantity || this.pricePerQuantity === 0) &&
      (!this.price || this.price === 0);
  }
}
